state/protocol/protocol_state/updater.go

```diff
@@ -187,17 +150,20 @@ func (u *Updater) ProcessEpochCommit(epochCommit *flow.EpochCommit) error {
 // No errors are expected during normal operations.
 func (u *Updater) UpdateIdentity(updated *flow.DynamicIdentityEntry) error {
```

Comment:
## ⚠️❓ 

I have a hard time convincing myself that this code is correct. Just as a brief sketch of my thoughts:
* at the point when we are leaving an epoch, the Identity Table (`EpochStateContainer.ActiveIdentities`) essentially snapshots the latest state when these nodes were active. 
  * In my opinion, changes of the dynamic identities essentially should only carry forward (causality) but not backwards. 
  * Hence, `ProtocolStateEntry.PreviousEpoch` should never be modified. But we do here: https://github.com/onflow/flow-go/blob/e76575031e516431e3592ac48c9c76d2902807b2/state/protocol/protocol_state/updater.go#L153-L156
* changing dynamic properties of node X affects the current and next epoch. However, as the weight might change at epoch boundaries, we cannot simply overwrite the weight for the current and next epoch by the same value. I think the following code is an incorrect simplification: https://github.com/onflow/flow-go/blob/e76575031e516431e3592ac48c9c76d2902807b2/state/protocol/protocol_state/updater.go#L157-L164

This is **not addressed in my PR**

Reply:
As I understand we will address it in subsequent PR